File: scripts/add-responsive-css.js
#!/usr/bin/env node
/**
 * Add responsive-portal-admin.css to all admin and portal pages
 */
const fs = require('fs');
const path = require('path');

const ROOT_DIR = process.cwd();
const CSS_LINK = '<link rel="stylesheet" href="/assets/css/responsive-portal-admin.css?v=mmp5r1rf">';

function addResponsiveCss(dir) {
    const files = fs.readdirSync(dir);
    let count = 0;
    
    for (const file of files) {
        const filePath = path.join(dir, file);
        try {
            const stat = fs.statSync(filePath);
            if (stat.isDirectory()) {
                count += addResponsiveCss(filePath);
            } else if (file.endsWith('.html') && !file.includes('node_modules')) {
                let content = fs.readFileSync(filePath, 'utf8');
                
                // Check if already has responsive-portal-admin.css
                if (content.includes('responsive-portal-admin.css')) {
                    continue;
                }
                
                // Find responsive-2026-complete.css and add after it
                const targetPattern = /(<link rel="stylesheet" href="\/assets\/css\/responsive-2026-complete\.css[^>]*>)/;
                if (targetPattern.test(content)) {
                    content = content.replace(targetPattern, `$1\n    ${CSS_LINK}`);
                    fs.writeFileSync(filePath, content, 'utf8');
                    count++;
                    console.log(`Added responsive CSS: ${filePath}`);
                } else {
                    // Fallback: add before </head>
                    const headPattern = /(<\/head>)/;
                    if (headPattern.test(content)) {
                        content = content.replace(headPattern, `    ${CSS_LINK}\n$1`);
                        fs.writeFileSync(filePath, content, 'utf8');
                        count++;
                        console.log(`Added responsive CSS (fallback): ${filePath}`);
                    }
                }
            }
        } catch (err) {
            // Skip errors
        }
    }
    
    return count;
}

console.log('Adding responsive-portal-admin.css to admin pages...');
const adminCount = addResponsiveCss(path.join(ROOT_DIR, 'admin'));

console.log('\nAdding responsive-portal-admin.css to portal pages...');
const portalCount = addResponsiveCss(path.join(ROOT_DIR, 'portal'));

console.log(`\nDone! Added responsive CSS to ${adminCount + portalCount} files.`);
console.log(`- Admin: ${adminCount} files`);
console.log(`- Portal: ${portalCount} files`);

File: scripts/add-responsive-css.py
#!/usr/bin/env python3
"""
Add responsive-enhancements.css to all HTML files
"""

import os
import re

BASE_DIR = "/Users/mac/.gemini/antigravity/scratch/sadec-marketing-hub"
EXCLUDE_PATTERNS = ["node_modules", "playwright-report", ".git", ".vercel"]

def add_responsive_css(file_path: str) -> tuple[bool, str]:
    """Thêm responsive-enhancements.css vào file HTML"""
    try:
        with open(file_path, 'r', encoding='utf-8') as f:
            content = f.read()

        # Kiểm tra đã có responsive-enhancements.css chưa
        if 'responsive-enhancements.css' in content:
            return False, "Đã có responsive-enhancements.css"

        # Tìm vị trí chèn (sau các link CSS khác trong head)
        # Tìm tất cả các link rel="stylesheet"
        stylesheet_links = list(re.finditer(r'<link[^>]*rel="stylesheet"[^>]*>', content))

        if stylesheet_links:
            # Chèn sau link stylesheet cuối cùng
            last_link = stylesheet_links[-1]
            insert_pos = last_link.end()

            # Tìm vị trí closing tag hoặc newline sau đó
            next_newline = content.find('\n', insert_pos)
            if next_newline != -1:
                insert_pos = next_newline + 1

            new_css_link = f'\n  <link rel="stylesheet" href="../assets/css/responsive-enhancements.css">'

            # Nếu path là root file (không trong subfolder)
            if file_path.count('/') == BASE_DIR.count('/') + 1:
                new_css_link = f'\n  <link rel="stylesheet" href="assets/css/responsive-enhancements.css">'

            content = content[:insert_pos] + new_css_link + content[insert_pos:]

            with open(file_path, 'w', encoding='utf-8') as f:
                f.write(content)

            return True, "Đã thêm responsive-enhancements.css"

        return False, "Không tìm thấy stylesheet links"

    except Exception as e:
        return False, f"Lỗi: {str(e)}"


def should_exclude(path: str) -> bool:
    return any(pattern in path for pattern in EXCLUDE_PATTERNS)


def main():
    html_files = []
    for root, dirs, files in os.walk(BASE_DIR):
        dirs[:] = [d for d in dirs if not should_exclude(d)]
        for file in files:
            if file.endswith('.html') and not should_exclude(root):
                html_files.append(os.path.join(root, file))

    print(f"Tìm thấy {len(html_files)} file HTML")
    print("=" * 60)

    added_count = 0
    for file_path in sorted(html_files):
        success, message = add_responsive_css(file_path)
        rel_path = file_path.replace(BASE_DIR + '/', '')[:55]

        if success:
            added_count += 1
            print(f"✅ {rel_path}")
        else:
            print(f"⏭️  {rel_path} - {message}")

    print("=" * 60)
    print(f"Đã thêm: {added_count}/{len(html_files)} files")


if __name__ == "__main__":
    main()
